config/smarty.config.inc.php

```diff
@@ -35,6 +35,7 @@
 }
 
 $smarty->setCompileDir(_PS_CACHE_DIR_.'smarty/compile');
+$smarty->setCompileId(_THEME_NAME_);
```

Comment:
```suggestion
$smarty->setCompileId(Context::getContext()->shop->id . '-' . _THEME_NAME_);
```
@eternoendless is right, it doesn't concern the multishop, combined to the theme name is more efficient